clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/StoreBufferService.java

```diff
@@ -65,6 +65,8 @@ public class StoreBufferService extends AbstractStoreBufferService {
 
   private final boolean isSorted;
 
+  private boolean isStarted = false;
```

Comment:
If there are multiple threads involved, then let's make this field volatile...

Reply:
looks like SBC creates all the drainer threads inside the `startInner` and it gets called once

Reply:
Sure, `isStarted` doesn't get called only from `startInner`, right?

Reply:
ah right! good catch